File: doonut-support/src/main/java/com/doonutmate/exception/ErrorResponse.java
package com.doonutmate.exception;

import org.springframework.http.HttpStatus;

public record ErrorResponse(
        int httpStatus,
        ErrorCode errorCode,
        String message,
        String path
) {

    public static ErrorResponse of(HttpStatus httpStatus, String message, String path) {
        return new ErrorResponse(httpStatus.value(), null, message, path);
    }

    public static ErrorResponse of(HttpStatus httpStatus, ErrorCode errorCode, String message, String path) {
        return new ErrorResponse(httpStatus.value(), errorCode, message, path);
    }
}

File: doonut-app-external-api/src/main/kotlin/com/doonutmate/oauth/exception/ErrorResponse.kt
package com.doonutmate.oauth.exception

import org.springframework.http.HttpStatus

data class ErrorResponse(
    val status: Int,
    val error: String,
    val message: String,
    val path: String,
) {
    constructor(httpStatus: HttpStatus, message: String, path: String) : this(httpStatus.value(), httpStatus.name, message, path)
}
